aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -0,0 +1,2326 @@
+
+        // actual coins on stake pool belonging to the active shares pool
+        active = active + pending_active;
+        // actual coins on stake pool belonging to the shares pool hosting `pending_inactive` stake
+        // at current observed lockup cycle, either pending: `pending_inactive` or already inactivated:
+        if (lockup_cycle_ended) {
+            // `inactive` on stake pool = any previous `inactive` stake +
+            // any previous `pending_inactive` stake and its rewards (both inactivated)
+            pending_inactive = inactive - pool.total_coins_inactive
+        };
+
+        // on stake-management operations, total coins on the internal shares pools and individual
+        // stakes on the stake pool are updated simultaneously, thus the only stakes becoming
+        // unsynced are rewards and slashes routed exclusively to/out the stake pool
+
+        // operator `active` rewards not persisted yet to the active shares pool
+        let commission_active = total_coins(&pool.active_shares);
+        commission_active = if (active > commission_active) {
+            multiply_then_divide(&pool.active_shares,
+                active - commission_active, pool.operator_commission_percentage, MAX_FEE)
+        } else {
+            // handle any slashing applied to `active` stake
+            0
+        };
+        // operator `pending_inactive` rewards not persisted yet to the pending_inactive shares pool
+        let commission_pending_inactive = total_coins(table::borrow(&pool.inactive_shares, pool.observed_lockup_cycle));
+        commission_pending_inactive = if (pending_inactive > commission_pending_inactive) {
+            multiply_then_divide(&pool.active_shares,
+                pending_inactive - commission_pending_inactive, pool.operator_commission_percentage, MAX_FEE)
+        } else {
+            // handle any slashing applied to `pending_inactive` stake
+            0
+        };
+
+        (lockup_cycle_ended, active, pending_inactive, commission_active, commission_pending_inactive)
+    }
+
+    /// Synchronize delegation and stake pools: distribute yet-undetected rewards to the corresponding internal
+    /// shares pools, assign commission to operator and eventually prepare delegation pool for a new lockup cycle.
+    public entry fun synchronize_delegation_pool(pool_address: address) acquires DelegationPool {
+        assert_delegation_pool_exists(pool_address);
+        let pool = borrow_global_mut<DelegationPool>(pool_address);
+        let (
+            lockup_cycle_ended,
+            active,
+            pending_inactive,
+            commission_active,
+            commission_pending_inactive
+        ) = calculate_stake_pool_drift(pool);
+
+        // zero `pending_active` stake indicates that either there are no `add_stake` fees or
+        // previous epoch has ended and should release the shares owning the existing fees
```

Comment:
Can you document that we're moving the fees from NULL_SHAREHOLDER and distribute that to the rest of the pool?